### PR TITLE
Update docs around Oracles

### PR DESCRIPTION
**Pull request by hushaohan**
### Summary

update documentation around Oracles, their circuit constructions, and related algorithms.

accompanying https://github.com/Qiskit/qiskit-aqua/pull/370

### CODE REVIEW

**Comment by hushaohan** [comment]
@chunfuchen could you please also add @pistoia as a reviewer?

**Review by pistoia — COMMENTED** [review]
Just a few comments:
1. This sentence: "Aqua includes utilities around circuit constructions simple logic expressions." in `algorithms.rst` line 165 is hard to parse.
2. Line 176: "each literal's absolute value indicates a corresponding variable" --> "each literal's absolute value indicates a variable"
3. "and a negative sign would indicate" --> "and a negative sign indicates"
4. In `oracles.rst`, line 126. "However, this oftentimes lead" --> "However, this oftentimes leads"
5. Line 128: "Aqua therefore provides" --> "Aqua, therefore, provides"
6. Line 170 and 171: perhaps we can remove the word "certain" (occurring twice)
7. In Section "Truth Table Oracle", I think that expressions n, m, n=1, m=2 etc. should be encoded in math mode (as opposed to code mode).

**Review by pistoia — APPROVED** [review]
